packages/devtools_app/lib/src/utils.dart

```diff
@@ -1069,3 +1069,225 @@ String prettyTimestamp(
   );
   return DateFormat.Hms().format(timestampDT); // HH:mm:ss
 }
+
+/// A [ChangeNotifier] that holds a list of data.
+///
+/// This class also exposes methods to interact with the data. By default,
+/// listeners are notified whenever the data is modified, but notifying can be
+/// optionally disabled.
+class ListValueNotifier<T> extends ChangeNotifier
+    implements ValueListenable<List<T>> {
+  /// Creates a [ListValueNotifier] that wraps this value [_rawList].
+  ListValueNotifier(List<T> rawList) : _rawList = List<T>.from(rawList) {
+    _currentList = ImmutableList(_rawList);
+  }
+
+  final List<T> _rawList;
+
+  ImmutableList<T> _currentList;
+
+  @override
+  List<T> get value => _currentList;
+  set value(List<T> newValue) {
```

Comment:
implement https://api.flutter.dev/flutter/foundation/ValueNotifier-class.html instead of implementing the ValueListenable API and then throwing.